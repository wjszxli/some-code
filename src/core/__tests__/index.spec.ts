import { createForm } from "..";

const testValues = { aa: 111, bb: 222 };

describe("createForm", () => {
  test("values", () => {
    const form = createForm({
      values: testValues,
    });
    expect(form.getFormState((state) => state?.values)).toEqual(testValues);
    expect(form.getFormState((state) => state?.initialized)).toEqual(true);
    expect(form.getFormGraph()).toMatchSnapshot();
  });

  test("initialValues on init", () => {
    const form = createForm({
      initialValues: testValues
    })
    const aa = form.registerField({ path: 'aa' })
    const bb = form.registerField({ path: 'bb' })

    expect(form.getFormState(state => state?.values)).toEqual(testValues)
    expect(form.getFormState(state => state?.initialValues)).toEqual(testValues)
    expect(form.getFormState(state => state?.initialized)).toEqual(true)
    expect(aa.getState(state => state?.value)).toEqual(testValues.aa)
    expect(bb.getState(state => state?.value)).toEqual(testValues.bb)
    expect(form.getFormGraph()).toMatchSnapshot()

  });
});
