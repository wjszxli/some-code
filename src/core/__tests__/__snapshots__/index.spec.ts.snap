// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createForm values 1`] = `
{
  "": {
    "displayName": "FormState",
    "editable": true,
    "errors": [],
    "initialValues": {},
    "initialized": true,
    "invalid": false,
    "loading": false,
    "modified": true,
    "mounted": false,
    "submitting": false,
    "unmounted": false,
    "valid": true,
    "validating": false,
    "values": {
      "aa": 111,
      "bb": 222,
    },
    "warnings": [],
  },
}
`;
